function/routes/archiveCreate.js

```diff
@@ -25,19 +28,59 @@ const validateInputs = (req) => {
   if (!_.isObject(args)) {
     throwHttpError(400, 'Request must contain a valid JSON body.');
   }
-  // verify non-empty workspace namespace, name, source, destination buckets
-  [argNamespace, argName, argSource, argDestination].forEach((key) => {
-    if (!nonEmptyString(args[key])) {
-      throwHttpError(400, `Request must contain a non-empty value for [${key}].`);
+  // verify non-empty source and destination workspaces
+  [argSource, argDestination].forEach((ws) => {
+    if (!args[ws]) {
+      throwHttpError(400, `Request must contain a value for [${ws}].`);
     }
+    [argNamespace, argName].forEach((key) => {
+      if (!nonEmptyString(args[ws][key])) {
+        throwHttpError(400, `Request must contain a non-empty value for [${ws}.${key}].`);
+      }
+    });
+  });
+
+  // construct & return arguments object.  The expected case is that this is
+  // equivalent to what the user input - but we strip an excesses and have validated it.
+  const userArgs = _.pick(args, ['source.namespace', 'source.name',
+    'destination.namespace', 'destination.name']);
+  return userArgs;
+};
+
+const validateWorkspaces = (appConfig, userArgs, token) => {
```

Comment:
nit - naming choice here.  I would expect a method named like this to return a `boolean` or maybe an array that's empty or populated with validation errors for the workspaces.